### ISSUE TITLE
Introduce new core install profiles for common uses (wordpress, portfolio, and more)

### ISSUE THREAD

**Issue by mikemccaffrey** [backdrop/backdrop-issues#3126]
Right now, we have a very one-size-fits-all experience out of the box with Backdrop, despite the fact that the standard install profile is not terribly applicable to anyone in particular. In order to show the flexibility of backdrop, and to make it easier for people to get started right away using backdrop for the purpose that they are intending, we might want to introduce several new core install profiles.

I'd recommend that we identify the primary use cases that people are looking to implement using Backdrop, and fashion some install profiles that we can include in core to get people going. 

One main one would be to create a wordpress-analog install profile that can be used in combination with our planned Wordpress to Backdrop migration #1055, that will set up all the content types, taxonomy, blocks, and such so that the content can be migrated straight in and look and work the same way as a standard wordpress site.

Another important install profile seems like it would be for a portfolio template, similar to what you can quickly build with squarespace. People should be able to download backdrop, install it with a portfolio profile, and get started right away plugging in their projects and experience.

**Comment by mikemccaffrey** [comment]
Just a note, but with this ticket, I'm trying to bypass some of the discussion around distributions that we've been stuck on (see #56 #99 #1175)  by proposing that we focus on creating install profiles that work with just the modules and features that we have in drupal core and don't require any external dependencies on contrib modules or themes.

**Comment by docwilmot** [comment]
I like this idea.

**Comment by jlfranklin** [comment]
I think this is a good idea, but I would caution against easily predictable pitfall: adding more modules into core simply because they're needed for profiles.  That's a sure path to core bloat.

**Comment by kreynen** [comment]
TL;DR There are a lot of issues with install profiles in Drupal.  Don't repeat those mistakes.

We're starting to build some sites using Backdrop as we continue to evaluate the Backdrop option at the University of Colorado.  We currently maintain all of our sites using https://www.drupal.org/project/express.  In a previous life, I created https://www.drupal.org/project/civicrm_starterkit which was featured as a default install option on Pantheon for years.  I am the primary maintainer of https://www.drupal.org/project/drupalorg_whitelist

That is my long-winded way of saying I have a lot of experience with install profiles.  

With 20/20 hindsight, I have some thoughts about mistakes we made in Drupal's approach to install profiles/distributions.  

I lobbied hard to keep Umami out of Drupal's core.  Mainly because there was really no reason that needed to be maintained in core other than the target audience had so little technical skill and was so fickle that even seeing a core install before Umami's improvements wouldn't allow them to evaluate Umami.

Umami is fundamentally flawed in that the install profile combines configuration and content in a way that it can't be used to build a different recipe related site.  If you try to keep Umami installed between major version issues like 8.4.x to 8.5.x, it will fatal error.

We designed packaged distributions to meet the needs of what we were building at the time.  If you go way back to https://archive.org/details/DrupalconSf2010DrupalDistributionsOnDrupal.org, we really thought we'd be able to build applications like OpenAtrium, starter kits like Commerce and CiviCRM and testing/tools using the same process and structure.

It turns out, this doesn't work very well.  At CU we've abstracted/extended Drupal's install profile to be more a foundation that each campus can customize with a core with optional hosting modules.  We have modules that create demo content, but that's not a requirement of the express.profile/foundation.  Boulder sets some configurations in our core that are requirements for every site.  Denver users a different configuration.  Each core defines the criteria for using different hosting modules.  This allows us to tweak the configuration when our code is installed on our own, on prem hosting, Pantheon, Travis or our local development environments.

The key to making this work is to keep the foundations basic on install. 
 We extend that by building bundles that provide the default configurations permissions, variables and settings for a collection of modules and custom code. 
 Many of these are just feature exports and a few alters... the time of work most developers are used to.  We then hide the option of enabling and configuring modules directly and limit site owners to only enable bundles through https://www.drupal.org/project/profile_module_manager

![cms-foundations-cores-bundles](https://user-images.githubusercontent.com/385234/42645772-2643813e-85bc-11e8-9b33-b122658f0d36.png)

Since install profiles aren't currently a big part of the Backdrop installs, I think there is an opportunity to fix the mistakes we made in Drupal without bloating or complicating core.

Because it's 2018, we don't need .make files to package modules into downloadable zips.  Modules, libraries and themes can be installed into their default location using the process Backdrop is using now.  We just need to be able to prepoulate what a foundation requires the user to install in their site.

Because Backdrop is focussed on a much simpler starting point, I don't see many advantages to injecting profiles into the install process.  Instead, what were profiles in Drupal could become a type of module labeled as foundations that mature organically.  While initially foundations would require more steps to the install process (Install Backdrop, download/enable Foundation and dependencies).  

If writing and maintaining foundations for Backdrop takes off and there are several stable, well maintained foundations to choose from, core could offer the option of downloading and installing a foundation as part of the install process.

Foundations can include an option to prepopulate demo content, but that should be uninstallable.  

While I can see some value to installing foundations with demo content in the default install process, the type of demo sites that exists for Wordpress themes like https://avada.theme-fusion.com/demos/ and a much more effective outreach tool for non-technical evaluators.

At CU, we wrote https://github.com/CuBoulder/cu_demo_bundle.  That bundle allows any user to login with admin level permissions to see how the site is configured, but then locks all forms so that content can't be added or configuration cahnges.  We've gotten a lot of possitive feedback about that from people who prefer learning by doing.

**Comment by jlfranklin** [comment]
@kreynen, you should check out [SilkscreenCMS](https://github.com/silkscreencms/silkscreencms), an enhanced version of Backdrop that lets you create [layered configurations](https://github.com/silkscreencms-contrib/config_layered).  Looking at your diagram, the Foundation release can host the base configuration, each of your Core layers maintains its configuration overrides on top of that base, and a local configuration store keeps the local settings.

**Comment by kreynen** [comment]
@jlfranklin I'll take a look, but one of the biggest concerns with have of going with Backdrop is the size of that community and whether or not we'll see enough benefit from the overhead modules that are generalized contrib vs. writing PHP specifically to meet our own needs.  We have enough resources to maintain a custom CMS, but no one wants to.  We'd rather be part of a community effort... just one that's more focused on building websites with low TCO.

**Comment by kreynen** [comment]
I'll also add that we don't really need install profiles anymore, but at the project level I think it is worth the time and effort maintaining install profiles for the same reasons it's worth keeping multisite.  We have a great DevOps solution at CU in https://cuboulder.github.io/atlas/.  We still install each Drupal instance using `drush si express -y` after prepopulating the settings.php files, but could just as easily import a preconfigured db for each site.

Multisite and install profiles probably don't meet the 80/20 criteria, but the advantage of having install profiles and multisite in Backdrop is it helps developers learn to work with Backdrop on a larger scale.   After building a few similar sites, a good developer will start looking at saving time by changing the way similar sites and built and hosting.  

Install profiles and multisite are also key features for higher ed which could very quickly make up a large percentage of Backdrop installs.

**Comment by klonos** [comment]
@kreynen :

> ... We currently maintain all of our sites using https://www.drupal.org/project/express. ...

...and from the description of that project:

> Site Owner != Site Builders

I have been [telling people that](https://github.com/backdrop/backdrop-issues/issues/1285#issuecomment-152736821), but they do not listen 😛

Thanks for all your input @kreynen 👍 ...and thanks @jlfranklin for forking Backdrop and maintaining Silkscreen (among other things). It really makes a point for us choosing to use GitHub as an open means of collaboration and for exploring the possibilities of new ideas 😄

**Comment by cellear** [comment]
I brought this up on Zulip, but I live @stpaultim's suggestion to note here what Joomla does: 
![image](https://user-images.githubusercontent.com/757296/103503725-0a0a0380-4e0a-11eb-800b-e6f74111f2b2.png)

**Comment by cellear** [comment]
The "Learn Joomla" option gives you this:

![image](https://user-images.githubusercontent.com/757296/103503749-2443e180-4e0a-11eb-8dd4-8483fd6c4fe4.png)

**Comment by philsward** [comment]
Trying to re-liven this issue, I think 3 profiles would be a good start:

1) New User How-To Profile
This profile is very tutorial based, designed to walk a new user through the important areas through hands on.

2) Sandbox Profile
A sandbox of random information, views, images etc that shows how various pieces of Backdrop work. A lot of people learn by reverse engineering. An idea is to make this a dynamic profile where the assets are downloaded at the time of install.

3) Clean Profile
This profile is for advanced users who want a clean slate with a few OOTB customizations so they can jump right into building a site without having to remove a bunch of pre-added items. This should be similar to a minimal profile, but have some of the common basics ready to go OOTB.

On a related note, I think "profiles" should become their own project type. This has already been brought up, but I want to touch on that idea for the big picture of how this can all come together.

**Comment by allsite** [comment]
Let's revisit this idea. Here's a discussion from zulip:

Obviously this is an opinion and perhaps others agree. I believe the following settings should default to "false" or "off", simply for sake of intentionality. I find it enjoyable to make clicks that enable functionality. In contrast, I find it annoying to have to repeatedly disable functions. If others agree a simple solution could be to default them differently or perhaps make a config page to allow developers to pick and choose how they default, e.g. "Content Type - Defaults > "Enable option to Promote by Default" while creating new content types.

User:
Notify user of new account (Imagine a "People" page, where they're user entities, but have yet to be assigned a business Role for site access. Essentially a database of people who may or may not play a publishing role. In this common scenario it's not useful to invite them to access the site at the time of account creation)

Content Types:
Preview before submitting
Show option for scheduling
Show option to promote
Show option to create new revisions

Views:
Link title to node/user page
Create a label
Use Pager
Items to display (10 default) (this one I actually enjoy, as it can prevent a potentially massive query before you have a chance to filter the results)
Auto Preview (I'm thankful this has been disabled by default, for similar performance reasons, nice job!)

Let's add the "Body" field to the list. I find it hard to believe the majority uses "Body with Summary" on the majority of their site's content types. For me personally it's target practice. Same with "Articles" and "Posts".

Modules like Comment or Color also feel bizarre in terms of being enabled by default.

The core/install.php?profile=minimal profile includes many of the defaults in question. In my opinion it omits a few items in a good way and omits others in a bad way, all while including most of the unwanted defaults as listed above.
